metals/src/main/scala/scala/meta/internal/metals/codeactions/OrganizeImports.scala

```diff
@@ -32,18 +30,20 @@ final class OrganizeImports(
     val uri = params.getTextDocument.getUri
     val file = uri.toAbsolutePath
     if (isSourceOrganizeImportCalled(params) && isScalaOrSbt(file)) {
-      ScalafixProvider.scalaVersionAndClasspath(file, buildTargets) match {
-        case Some((scalaVersion, scalaBinaryVersion, classpath))
-            if !ScalaVersions.isScala3Version(scalaVersion) =>
-          organizeImportsEdits(
-            file,
-            scalaVersion,
-            scalaBinaryVersion,
-            classpath
+
+      val scalaTarget = for {
+        buildId <- buildTargets.inverseSources(file)
+        target <- buildTargets.scalaTarget(buildId)
+      } yield target
+      scalaTarget match {
+        case Some(target)
+            if !ScalaVersions.isScala3Version(target.scalaVersion) =>
+          organizeImportsEdits(file, target)
+        case Some(target)
+            if ScalaVersions.isScala3Version(target.scalaVersion) =>
+          scribe.info(
+            s"Organize import doesn't work on ${target.scalaVersion} files"
```

Comment:
Do you think it's clear enough to the user that that this isn't supported in Scala 3 files? For example I triggered this in a Scala 3 file:

<img width="459" alt="Screenshot 2020-10-06 at 08 04 36" src="https://user-images.githubusercontent.com/13974112/95164686-ba844f00-07aa-11eb-914f-e76e5f977610.png">

And while we are logging this here that the rule isn't supported in this target version, do you think it might make sense to actually show this warning to the user to make it explicit why they are seeing the above message about the code action not being found?

Reply:
I think it's a bit different, since we are not adding the action at all and Metals is not expected to work in this case. Also, if someone has organize imports on save it would continuously show the message, because it will never work.

We can update it if it will be confusing for users.